src/python/pants/option/subsystem.py

```diff
@@ -86,14 +86,19 @@ class EnvironmentAware(metaclass=ABCMeta):
 
         To consume environment-sensitive options, inject the `EnvironmentAware` inner class into
         your rule.
+
+        Optionally, it is possible to specify environment variables that are required when
+        post-processing raw values provided by users (e.g. `<PATH>` special strings) by specifying
+        `env_vars_used_by_options`, and consuming `_options_env` in your post-processing property.
+        These environment variables will be requested at construction time.
         """
 
         subsystem: ClassVar[type[Subsystem]]
-        depends_on_env_vars: ClassVar[tuple[str, ...]] = ()
+        env_vars_used_by_options: ClassVar[tuple[str, ...]] = ()
 
         options: OptionValueContainer
         env_tgt: EnvironmentTarget
-        env_vars: EnvironmentVars = EnvironmentVars()
+        _options_env: EnvironmentVars = EnvironmentVars()
```

Comment:
Only thing I'm not certain on is if this should be public or private. Is this to avoid clashes with options? Seems unlikely someone will have `--options-env`

Reply:
It's meant to be private -- the value is meant to be consumed in post-processing properties for options (e.g. in `ShellSetup.EnvironmentAware.executable_search_path`, and not as a general way to request environment variables alongside the subsystem. Obviously this is Python and we can't stop people from accessing the value, but communicating intent is important.